requirements.txt

```diff
@@ -5,5 +5,5 @@ requests>=2.18.4,<3.0
 PyNaCl>=1.1.2,<2.0
 pymacaroons>=0.12.0,<1.0
 six>=1.11.0,<2.0
-protobuf>=3.4.0,<4.0
+protobuf>=4.21.1,<=4.21.12
```

Comment:
What about``protobuf>=3.4.0,<5.0`` instead? If another dependency (e.g. python-mysql-connector) needs something more specific but within range, the pip resolver will take care of that.

We intend to use python-libmaas with Openstack Watcher but we're having issues with the upper constraints as it tries to use 4.24.3:

https://github.com/openstack/requirements/blob/e54dc19f590bd0ea74fd2d7ada3d0d1b3321260b/upper-constraints.txt#L377
https://review.opendev.org/c/openstack/watcher/+/898790

```
The conflict is caused by:
    macaroonbakery 1.3.1 depends on protobuf<4.0 and >=3.0.0
    The user requested (constraint) protobuf===4.24.3
```

Reply:
I see your conflict, yeah I think it's a good suggestion, it might work for everyone 👍

Reply:
Yeah I'm not sure why I gave this an upper bound anyways, you're right pip should take care of the specific boundaries, thanks for the suggestion @petrutlucian94!